azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/tools/auth/core/DeviceCodeCredentialRetriever.java

```diff
@@ -0,0 +1,30 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.tools.auth.core;
+
+import com.azure.identity.DeviceCodeCredential;
+import com.azure.identity.DeviceCodeCredentialBuilder;
+import com.microsoft.azure.AzureEnvironment;
+import com.microsoft.azure.tools.auth.model.AuthMethod;
+import com.microsoft.azure.tools.auth.model.AzureCredentialWrapper;
+import com.microsoft.azure.tools.common.util.TextUtils;
+import org.apache.commons.lang3.StringUtils;
+
+public class DeviceCodeCredentialRetriever extends AbstractCredentialRetriever {
+    private static final String AZURE_TOOLKIT_CLIENT_ID = "777acee8-5286-4d6e-8b05-f7c851d8ed0a";
```

Comment:
We may need to make AZURE_TOOLKIT_CLIENT_ID  a parameter for different tools

Reply:
add as TODO items, since maven plugin will not need to change this.